### PR TITLE
Basic UI: several fix for selection and switch with mappings

### LINKED ISSUE #3143: In Sitemaps LabelColor and ValueColor not working on Selection elements in BasicUI
I've defined the following switch:

`Switch hmvar_Presence "Presence" <parents_3_3> (gPresence) {channel="homematic:GATEWAY-EXTRAS-CCU2:ccu2:GWE00000000:1#Presence"}`

In my sitemap I tried to apply the following label colors:

`Selection item=PresenceString label="Presence" labelcolor=[hmvar_Presence==ON="green", hmvar_Presence==OFF="red"] mappings=[auto="auto", present="here", notPresent="not here"]`

The entry is shown without any labelcolor in the BasicUI.

It works perfectly in the ClassicUI, except that not only the label is colored, but the displayed selection entry, too. Though this might be by design (and is okay for me).

### LINKED ISSUE #3346: Basic UI: value not updated for a selection element in sitemap
I have a selection item associated to a string item.
When I open the page containing this element, the value is correct.
Then if the value is updated, the selection element displays a blank value.

### LINKED ISSUE #3349: Basic UI: value at left of buttons not refreshed for switch element with mappings
This issue concerns the switch elements with mappings in sitemap.
The value at left of buttons is correctly displayed when opening the page but then not refreshed.
Buttons state are correctly refreshed.

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #3143
Fix #3346
Fix #3349

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@resetnow : that is for you ;)

File: extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js [hunk 1]
```diff
@@ -478,12 +479,11 @@
 					item: _t.item,
 					value: value
 			}));
-			_t.suppressUpdate();
```
Comment (resetnow) [inline extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js]: [1] I understand that you need to obtain some properties from server so you don't want to suppress the callback entirely...

File: extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js [hunk 2]
```diff
@@ -494,12 +494,16 @@
 			_t.reset();
```
Comment (resetnow) [inline extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js]: [2] ...but could you please implement another way to suppress the useless call to `reset()`?

File: extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js [hunk 3]
```diff
@@ -455,7 +455,8 @@
 		var
 			_t = this;
 
-		_t.value = _t.parentNode.querySelector(o.formValue);
+		_t.hasValue = _t.parentNode.getAttribute("data-has-value") === "true";
+		_t.value = _t.parentNode.parentNode.querySelector(o.formValue);
 		_t.count = _t.parentNode.getAttribute("data-count") * 1;
```
Comment (resetnow) [inline extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js]: [3] For example, `_t.suppressReset = false` here.

File: extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js (hunk 2 above)
Comment (resetnow) [inline extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js]: [4]

```
if (_t.suppressReset) {
	_t.suppressReset = false;
	return;
} else {
	_t.reset();
}
```

Comment (resetnow) [inline extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js]: [5] This has to do with animations and avoiding touching DOM too much

File: extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js (hunk 1 above)
Comment (resetnow) [inline extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js]: [6] `_t.suppressReset = true`

**Comment by lolodomo** [comment]
@resetnow : thank you for your quick review, I fixed what you requested.

File: extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js [hunk 4]
```diff
@@ -478,28 +480,37 @@
...
+		_t.setValuePrivate = function(value, itemState) {
+			if (_t.value !== null && _t.hasValue) {
 				_t.value.innerHTML = value;
 			}
 
 			if (_t.count === 1) {
 				return;
 			}
 
+			if (_t.suppressUpdateButtons ) {
```
Comment (resetnow) [inline extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js]: (space here)

**Comment by lolodomo** [comment]
Space suppressed.

**Comment by sjka** [comment]
Thanks!

**Pull request merged**